src/types/ant.ts

```diff
@@ -0,0 +1,165 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { z } from 'zod';
+
+import { Logger } from '../common/logger.js';
+import { AoMessageResult, WalletAddress, WriteOptions } from './common.js';
+
+export type AoANTState = {
+  Name: string;
+  Ticker: string;
+  Denomination: number;
+  Owner: WalletAddress;
+  Controllers: WalletAddress[];
+  Records: Record<string, AoANTRecord>;
+  Balances: Record<WalletAddress, number>;
+  Logo: string;
+  TotalSupply: number;
+  Initialized: boolean;
+  ['Source-Code-TX-ID']: string;
+};
+
+export type AoANTInfo = {
+  Name: string;
+  Owner: string;
+  Handlers: string[];
+  ['Source-Code-TX-ID']: string;
+  // token related
+  Ticker: string;
+  ['Total-Supply']: string;
+  Logo: string;
+  Denomination: string;
+};
+
+export type AoANTRecord = {
+  transactionId: string;
+  ttlSeconds: number;
+};
+
+export interface AoANTRead {
+  getState(): Promise<AoANTState>;
+  getInfo(): Promise<AoANTInfo>;
+  getRecord({ undername }): Promise<AoANTRecord | undefined>;
+  getRecords(): Promise<Record<string, AoANTRecord>>;
+  getOwner(): Promise<WalletAddress>;
+  getControllers(): Promise<WalletAddress[]>;
+  getTicker(): Promise<string>;
+  getName(): Promise<string>;
+  getBalance({ address }: { address: WalletAddress }): Promise<number>;
+  getBalances(): Promise<Record<WalletAddress, number>>;
+}
+
+export interface AoANTWrite extends AoANTRead {
+  transfer(
+    { target }: { target: WalletAddress },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  addController(
+    {
+      controller,
+    }: {
+      controller: WalletAddress;
+    },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  removeController(
+    {
+      controller,
+    }: {
+      controller: WalletAddress;
+    },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  setRecord(
+    {
+      undername,
+      transactionId,
+      ttlSeconds,
+    }: {
+      undername: string;
+      transactionId: string;
+      ttlSeconds: number;
+    },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  removeRecord(
+    { undername }: { undername: string },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  setTicker(
+    { ticker }: { ticker: string },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+  setName(
+    { name }: { name: string },
+    options?: WriteOptions,
+  ): Promise<AoMessageResult>;
+}
+
+export const AntRecordSchema = z
+  .object({
+    transactionId: z.string(),
+    ttlSeconds: z.number(),
+  })
+  .passthrough();
```

Comment:
zod allows inferring types from schemas - so we should replace our existing types with ones inferred on the zod schemas for ANT specific types in this PR. 

i.e. replace `AoANTRecord` type above with 

```typescript
export type AoANTRecord = z.infer(typeof AntRecordSchema)
```

Reply:
ME LIKEY